typescript/sdk/src/token/EvmERC20WarpModule.ts

```diff
@@ -74,7 +81,64 @@ export class EvmERC20WarpModule extends HyperlaneModule<
     TokenRouterConfigSchema.parse(expectedConfig);
     const actualConfig = await this.read();
 
-    return this.updateIsm(actualConfig, expectedConfig);
+    const transactions = [];
+
+    transactions.push(
+      ...(await this.updateIsm(actualConfig, expectedConfig)),
+      ...(await this.updateRemoteRouters(actualConfig, expectedConfig)),
+    );
+
+    return transactions;
+  }
+
+  /**
+   * Updates the remote routers for the Warp Route contract.
+   *
+   * @param actualConfig - The on-chain router configuration, including the remoteRouters array.
+   * @param expectedConfig - The expected token router configuration.
+   * @returns A array with a single Ethereum transaction that need to be executed to enroll the routers
+   */
+  async updateRemoteRouters(
+    actualConfig: TokenRouterConfig,
+    expectedConfig: TokenRouterConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
```

Comment:
the array type is useful for the abstract update() func, but since this is quite specific to the warp module and we're always going to return just one transaction, I'm not opposed to it returning `Promise<AnnotatedEV5Transaction>` instead

Reply:
I actually prefer the array. imho, it's an additional mental load to remember that it's a single array, but i think it reduces the codesize in this case

1. If we use an object, we have to return an empty object in one of the branches (see Noah's [suggestion](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4145/files/dbdb18235322a3cb09a40aaa06a9bf6df8525240#r1680195452)), which means the calling function will have to handle the empty object. In this case, we will have an empty object in the [transactions](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/dbdb18235322a3cb09a40aaa06a9bf6df8525240/typescript/sdk/src/token/EvmERC20WarpModule.ts#L86-L89)
2. If we keep it as an array, we can spread an empty array, which results in a noop.

Reply:
can we have 2 fns , `updateRemoteRouters(): tx[]` and `updateRemoteRouter: tx` where the first calls into the second, and the caller can decide whichever they'd like to call?

re codesize: not sure we should index on this outside of contract development (iff additional code == better readability+maintainability)

Reply:
Driveby, but agree with @ltyu, a single function that returns arrays is simplest. Definitely disagree that additional code is better readability (all things being equal)

Reply:
im going to keep it as is.

these are helper functions specifically for the `update()` and the interface for each of the current and future ( e.g., hook updates) will be the same, which is that they all return arrays.